### DISCUSSION TITLE
How to set height of textbox？it will change automatically

### DISCUSSION THREAD

**Discussion by webtj** [fabricjs/fabric.js#8061] [Q&A] [answered]
```js
//Example 1
let textInstance = new fabric.Textbox('this is textbox', {
    left: 120,
    top: 120,
    opacity: 1,
    borderRadius: 0,
    angle: 0,
    width: 120,
    fontSize: 16,
    fill: '#000000',
   height:200
})
canvasEditor.add(textInstance).renderAll()

//Example 2
let textInstance = canvasEditor.getObjects()[0]
textInstance.set({
  height:200
})
```
Example 1：height attribute does not take effect
Example 2:  The height of the textbox on the canvas is 200，But when I modify the text content or change the alignment, the height automatically becomes the line height of the text content。

The application scenarios are as follows
For a background image, you need to add a textbox with the same width and height as the background image, and then set textAlign to justify, then the text will be centered relative to the image.

is there any way？

please help me！please

**Comment by FabianElsmer** [comment, marked as answer]
I think a possible approach to this would be to create a `Group` first, then add both the `Textbox` and a `Rect` to it.

The `Rect` would then only contain the background image, and could be sized as desired, the `Textbox` only shows the text and should get the appropriate `width` set

**Comment by webtj** [comment]
> For a background image, you need to add a textbox with the same width and height as the background image, and then set textAlign to justify, then the text will be centered relative to the image.

Increased complexity of layer operations，

> I think a possible approach to this would be to create a `Group` first, then add both the `Textbox` and a `Rect` to it.
> 
> The `Rect` would then only contain the background image, and could be sized as desired, the `Textbox` only shows the text and should get the appropriate `width` set

this approach increases the complexity of layer operations and saves irrelevant group and rect data to the database, but it is theoretically feasible, I can give it a try.
thank you

**Comment by alejandro28100** [comment]
@webtj  I'm trying to achieve the same functionality. Please let me know if you solved this and how.

**Reply by webtj** [reply]
yes，refer to the answer above.

create a Group first, then add both the Textbox and a Rect to it.  

Change the properties of React to simulate the desired effect.
